### PR TITLE
updating the table of contents styles

### PR DESCRIPTION
**Pull request by susanev**
## Description
this is part of https://github.com/pulumi/docs/issues/8980
only includes styles for on this page; will tackle the rest in another pr

i need to open a registry pr to go along with this
registry pr: https://github.com/pulumi/registry/pull/2487

@anita-trimbur a few open questions for ya..
- not sure i got the white space correct; pls let me know if you want changes
- we need more indentation for h3s i think; but i didn't see it in ur designs; let me know what you want to do

Screenshot 2023-05-15 at 9 46 25 AM

Screenshot 2023-05-15 at 9 56 26 AM
Screenshot 2023-05-15 at 9 56 06 AM
Screenshot 2023-05-15 at 9 46 25 AM

this should merge after the docs nav work

## Checklist:

- [x] I have reviewed the [style guide](https://github.com/pulumi/pulumi-hugo/blob/master/STYLE-GUIDE.md).
- [x] If blogging, I have reviewed the [blogging guide](https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md).
- [ ] I have manually confirmed that all new links work.
- [x] I added aliases (i.e., redirects) for all filename changes.
- [ ] If making css changes, I rebuilt the bundle.

### CODE REVIEW

**Comment by anita-trimbur** [comment]
from figma
- [ ] h3s in the toc should be indented 16px

File: themes/default/layouts/partials/docs/right-nav.html [hunk 1]
```diff
@@ -1,12 +1,15 @@
 {{ with .File }}
     {{ if not $.Page.Params.no_on_this_page }}
-        <div class="on-this-page">
-            <h4 class="no-anchor">
-                <a href="#">On This Page</a>
-            </h4>
-            <ul class="text-sm">
-                {{/* This will be populated dynamically with all H2s and H3s on the page; see assets/js/main.js */}}
-            </ul>
+        <div class="on-this-page table-of-contents">
```
Comment (sean1588) [inline themes/default/layouts/partials/docs/right-nav.html:3]: do we need this `on-this-page` class anymore? it looks like we removed/renamed it. if not ignore me.

**Review by sean1588 — APPROVED** [review]
works great! :shipit:

**Comment by anita-trimbur** [comment]
do we disable the toc for top level pages? like the pulumi cloud page has multiple headings, but i don't see this component show up there

**Review by anita-trimbur — APPROVED** [review]
styles and behavior lgtm!!

**Review by anita-trimbur — APPROVED** [review]
styles and behavior lgtm!!

**Comment by susanev** [comment]
@anita-trimbur that might be my fault on the cloud page; let me take a look

File: themes/default/layouts/partials/docs/right-nav.html (hunk 1 above)
Comment (susanev) [reply themes/default/layouts/partials/docs/right-nav.html:3]: @sean1588 this is actually used in the ts file 🙃 
ive cleaned that up though in my incoming commit

**Pull request merged**